signac/contrib/migration/v1_to_v2.py

```diff
@@ -15,10 +15,10 @@
 
 import os
 
+from signac._synced_collections.backends.collection_json import BufferedJSONAttrDict
```

Comment:
Maybe we should replace all references to `signac._synced_collections.backends.collection_json.BufferedJSONAttrDict` with `signac.JSONDict`, which is an alias that we publicly expose. Then the dependence on synced collections is more isolated.

Reply:
We also use `JSONAttrDict` in some parts of our code. Since I am not super familiar with the code I am not sure if it's safe to replace the use of `JSONAttrDict` with `BufferedJSONAttrDict` (as `signac.JSONDict` is a `BufferedJSONAttrDict`)

Reply:
I say let's leave this as is for now. If and when we get around to making `synced_collections` a separate published package it'll get easier since we'll be referencing an external, public package.